include/deal.II/base/polynomial.h

```diff
@@ -807,6 +817,141 @@ namespace Polynomials
 
 
 
+  template <typename number>
+  template <typename Number2>
+  inline void
+  Polynomial<number>::value(const Number2      x,
```

Comment:
As far as I can see this method only has been moved into the header file duet to the new template argument.

Reply:
Yes, it was moved from there, I adjusted the types of temporary variables to `Number2` and removed this shortcut https://github.com/dealii/dealii/blob/d949861f2fb94acc0ce0f318983e214f87913c34/source/base/polynomial.cc#L216-L223 The reason for removing it is that (a) this is the unlikely path (almost all polynomials we use are in the Lagrange product form), so I do not think we need to optimize for it, and (b) I do not want to template the `value(const number x)` function (otherwise we lose the implicit conversion from `int` to `double`, for example) and do not see a reason to implement yet another specialization.